setup-commit-signing/action.yml

```diff
@@ -1,19 +1,46 @@
 name: setup-commit-signing
-description: Set up GPG commit signing
+description: Set up SSH commit signing
 author: nandahkrishna
 branding:
   icon: user-check
   color: green
 inputs:
   signing_key:
-    description: Secret key to use for signing commits
+    description: SSH or GPG private key to use for signing commits
     required: true
+  ssh:
+    description: Whether to do SSH or GPG signing
+    required: false
+    default: "false"
 runs:
   using: composite
   steps:
-    - run: |
+    - name: Enable SSH agent
+      if: inputs.ssh == 'true'
+      run: |
+        eval $(ssh-agent)
+
+        echo "SSH_AUTH_SOCK=${SSH_AUTH_SOCK}" >> ${GITHUB_ENV} # zizmor: ignore[github-env]
+        echo "SSH_AGENT_PID=${SSH_AGENT_PID}" >> ${GITHUB_ENV} # zizmor: ignore[github-env]
```

Comment:
NB: This `GITHUB_ENV` writes are safe always safe (single-line), assuming `ssh-agent` itself produces the expected output.